src/common/eeprom.cpp

```diff
@@ -409,63 +413,115 @@ static eeprom_vars_t &eeprom_startup_vars() {
     return ret;
 }
 
+/**
+ * @brief reads eeprom record from RAM structure into variant variable
+ *
+ * @param id eeprom record index
+ * @return variant8_t eeprom record
+ */
 variant8_t eeprom_get_var(enum eevar_id id) {
-    uint16_t addr;
-    uint16_t size;
-    uint16_t data_size;
-    void *data_ptr;
     variant8_t var = variant8_empty();
     if (id < EEPROM_VARCOUNT) {
-        eeprom_lock();
         var = variant8_init(eeprom_map[id].type, eeprom_map[id].count, 0);
-        size = eeprom_var_size(id);
-        data_size = variant8_data_size(&var);
-        if (size == data_size) {
-            addr = eeprom_var_addr(id);
-            data_ptr = variant8_data_ptr(&var);
-            st25dv64k_user_read_bytes(addr, data_ptr, size);
+        eeprom_get_var(id, variant8_data_ptr(&var), variant8_data_size(&var));
```

Comment:
note:
id < EEPROM_VARCOUNT is checked twice. Once here and second time inside eeprom_get_var(id, variant8_data_ptr(&var), variant8_data_size(&var));
I don't know if this can be meaningfully removed - probably not.
As other places from where eeprom_get_var(id, variant8_data_ptr(&var), variant8_data_size(&var)) is called doesn't check id before. Here is the check very probably necessary because of variant8_init() call.